.github/workflows/test_build.yaml

```diff
@@ -0,0 +1,61 @@
+name: test_build
+on:
+  pull_request:
+    branches:
+      - main
+    tags:
+      - "*"
+  push:
+    branches:
+      - main
+    tags:
+      - "*"
+
+# unfortunately, only one job per runner is possible
+# https://github.com/orgs/community/discussions/26769
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+env:
+  CONDA: /astro/users/kbmod/miniconda3
+
+jobs:
+  test-legacy:
+    name:  Python ${{ matrix.python-version }}
+    runs-on: self-hosted
+    strategy:
+      fail-fast: true
+      matrix:
+        python-version: ["3.8", "3.9", "3.10"]
+    defaults:
+      run:
+        shell: bash -leo pipefail {0}
+    steps:
+      - name: Setup environment
+        run: echo "/usr/local/cuda-11.8/bin" > $GITHUB_PATH
+        # setup-conda can only cache what setup-conda installs,
+        # we install via pip so we're re-downloading many of
+        # the same packages. This is good, but it's slow.
+      - name: Setup miniconda
+        uses: conda-incubator/setup-miniconda@v2.2.0
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          submodules: recursive
+        # Run linters before attempting to compile and install
+        # the package, faster fails.
+      - name: Linter
+        uses: psf/black@stable
+        with:
+          src: "./src"
```

Comment:
Do we want to extend this to the tests/ and src/analysis/ directories?

Reply:
Added the directories to the linter check. Ran black on them to get them fixed - I put that in a separate commit if you want to revert the changes, but note that because of the fail-fast strategy no tests will pass until we get them to be compliant.